Apply Error Code to catch & response error properly

13circle opened the pull request:
resolve #13

13circle committed:
Feat: Add ErrorCode schema

13circle committed:
Feat: Add init-error-code API at admin

13circle committed:
Feat: Serialize error response

13circle committed:
Refactor: Adopt ErrorCode to middlewares

13circle committed:
Feat: Warn if there are no error codes in DB

13circle committed:
Chore: Add explanation to init-error-code API

13circle committed:
Fix: Remove redundant ObjectId validation

13circle committed:
Refactor: Adopt ErrorCode to auth APIs - 1

- GET /api/auth/user-list
- GET /api/auth/user-detail/:id
- PATCH /api/auth/modify

ndaemy commented on src/models/errorCode.ts:
```diff
@@ -0,0 +1,20 @@
+  errorCode: { type: String, required: true },
+  httpStatus: { type: Number, required: true },
+  message: { type: String, required: true },
+});
```
에러코드를 아예 db에 넣어버리셨구만

13circle replied:
네넵 DB설때 이렇게 하길래 어설프게나마 따라해보았습니다.

ndaemy commented on src/api/admin/admin.ctrl.ts:
```diff
@@ -57,18 +57,13 @@ export const toAdmin = async (ctx: RouterContext): Promise<void> => {
     await admin.save();
 
-    ctx.body = user.serialize();
+    ctx.body = user?.serialize();
```
이건 이렇게 쓰면 뻑날 가능성이 있기 때문에, user를 if문으로 없을 경우를 처리하고 나서 ?를 빼는게 좋아요

13circle replied:
미들웨어에서 걸러줘서 상관은 없어보입니다만 그래도 할까요?

ndaemy replied:
그런거라면 ! 고고

13circle committed:
Feat: Handle generating error codes

13circle committed:
Refactor: Handle error w/ errorGenerator - 1

noah0316 committed:
Refactor: Adopt ErrorCode to auth & admin APIs
- POST /api/auth/register
- POST /api/auth/login
- GET /api/auth/check
- POST /api/auth/email-check/:id/:token
- POST /api/admin/approve/:id

noah0316 committed:
Refactor: Adopt errGenerator.ts to middlewares
- checkAdmin.ts
- checkLoggedIn.ts
- checkObjectId.ts
